cms/forms.py

```diff
@@ -16,3 +21,85 @@ def __init__(self, data=None, files=None, parent_page=None, *args, **kwargs):
             self.fields["overrides"].block.child_blocks["course_run"].child_blocks[
                 "readable_id"
             ].parent_readable_id = parent_page.specific.course.readable_id
+
+
+class CoursePageForm(WagtailAdminPageForm):
+    """
+    Admin form for CoursePage and ExternalCoursePage.
+
+    This form introduces price and course_run fields to manage pricing in cms.
+    """
+
+    price = forms.DecimalField(required=False)
+    course_run = forms.ChoiceField(required=False)
+
+    def __init__(self, data=None, files=None, parent_page=None, *args, **kwargs):
+        """
+        Adds choices for course_run field.
+        """
+        super().__init__(data, files, parent_page, *args, **kwargs)
+
+        instance = kwargs.get("instance", None)
+        if instance and instance.id:
+            course_runs = instance.course.courseruns.all()
+            course_run_choices = [("", "")] + [(run.id, run) for run in course_runs]
+            self.fields["course_run"].choices = course_run_choices
+
+    def save(self, commit=True):
+        """
+        Handles pricing update and creates product(if required) and product version for a course run.
+        """
+        page = super().save(commit=False)
+
+        course_run_id = self.cleaned_data["course_run"]
+        price = self.cleaned_data["price"]
+        if course_run_id and price:
+            course_run = CourseRun.objects.get(id=course_run_id)
+            product, _ = Product.objects.get_or_create(
+                content_type=ContentType.objects.get_for_model(CourseRun),
+                object_id=course_run.id,
+            )
+            ProductVersion.objects.create(product=product, price=price)
```

Comment:
While we were manually creating the products and production versions we had some additional fields e.g. take a look at [this product on RC](https://rc.xpro.mit.edu/admin/ecommerce/product/121/change/) you will notice:

1. Option to toggle a product `Active`. (Should we not give this option to Course teams)
2. Option to make a product `Public` or `Private`

And If you take a look at a product version e.g. look at [this product version on RC](https://rc.xpro.mit.edu/admin/ecommerce/productversion/183/change/) you will notice that there is:
1. An option to specify if a product needs enrollment code.

@cachob What do suggest about above concerns?

Reply:
I haven't made this change. @cachob What are your thoughts?